curl.ml

```diff
@@ -567,9 +569,22 @@ type pauseOption = PAUSE_SEND | PAUSE_RECV | PAUSE_ALL
 
 external pause : t -> pauseOption list -> unit = "caml_curl_pause"
 
+external writefunc_pause : unit -> int = "caml_curl_writefunc_pause" 
+
+let writefunc_pause = writefunc_pause ()
+
 let set_writefunction conn closure =
   setopt conn (CURLOPT_WRITEFUNCTION closure)
 
+let set_writefunction_opt conn (closure:curlWritefuncOpt) =
+  let closure s =
+    match closure s with
+      | `Full -> String.length s
+      | `Partial n -> n
```

Comment:
I don't think Partial is useful, it will be an abort for libcurl anyway
```
If that amount differs from the amount passed  to  your  callback  function,
       it'll  signal  an  error  condition  to  the  library.  This  will  cause  the  transfer  to  get  aborted and the libcurl function used will return
       CURLE_WRITE_ERROR.
```